### ISSUE TITLE
User/group ID for folders on host

### ISSUE THREAD

**Issue by schnillerman** [qvicksilver/docker-logitechmediaserver#4]
Hi,

I'm desperately trying to set this container up on a Synology NAS.

In order to match the container user/group to a user with access to host folders which I want to mount inside the container, I want to use existing host priviliges.

Therefore, I need to set the corresponding env variables.

I tried the following variables:

- PUID/PGID
- user_id/group_id

Values were (for testing purposes): 0/0 (which supposedly equals root)

None of them worked - the only way to get it to work is to give "everyone" permission to RW or R to the folders outside the container. This is not what I want, though.

Is there any way I can do better?

**Comment by qvicksilver** [comment]
Not really sure what you are doing on your side, but I run it as follows:
My folders are owned by user 1000 and I simply supply 1000 as the first argument for the container entrypoint script.

ls -ld /opt/storage/music
drwxr-xr-x 1 1000 1000 44 Feb  4  2017 /opt/storage/music

And the container is started like this:

docker run -d -p 9000:9000 -p 3483:3483 -p 3483:3483/udp -v /opt/storage/music:/opt/storage/music qvicksilver/logitechmediaserver 1000

**Comment by schnillerman** [comment]
I'm sorry I might've been a little unclear about my problem.

The docker container needs access to a folder on the host. Therefore, I think I have 2 choices:

1. Give "everyone" access to host folder so that the docker container and whichever user is defined inside can access it
2. Tell the docker container which host user/group ID to map with the container internal user

 I would prefer option 2, since I would like to keep permissions on host folders as-is.

Therefore, I would need to somehow transfer the desired host user/group ID to the container, as far as I understand, this might be possible by means of env variables. In most cases (on most forums), these env variables are PUID (user) and PGID (group).

Example: host user admin has PUID 1024 and PGID 101 by standard, I think. If I set the corresponding docker container's env variables accordingly, the docker would impersonate the admin user for the container.

As far as I can see, setting these env variables in your container does not have any effect.

See https://www.reddit.com/r/docker/comments/9m52qe/a_possible_solution_for_docker_volumes/ for a very similar problem description.

Was I able to make myself a little clearer?

Regards

**Comment by qvicksilver** [comment]
Absolutely! My container does not take user/group arguments as environment variables. Instead the first command line argument to the entrypoint script is the UID of the user to run the server as.
There is currently no way to set the GID.
So in my example above the directory is owned by UID 1000 and the server runs as UID 1000 and things work as you would expect.

Does that solve things for you?

**Comment by schnillerman** [comment]
Thanks for taking time to reply! Well, honestly, it doesn't solve things for me, but after a little further digging, this article describes my problem pretty well and seems to provide a solution:
https://denibertovic.com/posts/handling-permissions-with-docker-volumes/
And this article, too:
https://docs.docker.com/engine/security/userns-remap/
I am now inclined to try my own LMS docker file combining the contents of the two articles.
